portality/forms/application_processors.py

```diff
@@ -301,12 +301,16 @@ def finalise(self, account, save_target=True, email_alert=True):
 
         if self.source is None:
             raise Exception(Messages.EXCEPTION_EDITING_NON_EXISTING_APPLICATION)
-        if self.source.current_journal == constants.APPLICATION_STATUS_ACCEPTED:
+
+        if self.source.application_status == constants.APPLICATION_STATUS_ACCEPTED:
             raise Exception(Messages.EXCEPTION_EDITING_ACCEPTED_JOURNAL)
+
         if self.source.current_journal is not None:
```

Comment:
* what is difference between `related_journal_object` and `current_journal`? 
* should we following logic in frontend `portality/templates/application_form/_application_warning_msg.html` use `related_journal_object`?

Reply:
> what is difference between related_journal_object and current_journal?

`current_journal` is a field which contains the identifier of a journal *if the application is an active update request that's being reviewed by editorial*

`related_journal` is a field which contains the identifier of a journal *if the application is a completed update request that has been accepted or rejected*

`related_journal_object` gets the `models.Journal` object for the journal identified in `related_journal`.  I've added it as a function because it is necessary to access that data sometimes via the jinja2 templates, and that makes it easy.

> should we following logic in frontend portality/templates/application_form/_application_warning_msg.html use related_journal_object?

In this case this is the correct thing to do, because the update request is not accepted, so it will either be rejected or have a current_journal, and we don't need to show an error message if the journal is rejected, because editing it makes no difference.  So I think this is fine.